dataplane/src/packet_meta.rs

```diff
@@ -0,0 +1,125 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Open Network Fabric Authors
+
+use routing::vrf::VrfId;
+use std::collections::HashMap;
+
+#[derive(Debug, Default)]
+pub struct InterfaceId(u32);
+#[allow(unused)]
+impl InterfaceId {
+    pub fn get_id(&self) -> u32 {
+        self.0
+    }
+    pub fn set_id(&mut self, id: u32) {
+        self.0 = id
+    }
+}
+
+#[derive(Debug, Default)]
+pub struct BridgeDomain(u32);
+#[allow(unused)]
+impl BridgeDomain {
+    pub fn get_id(&self) -> u32 {
+        self.0
+    }
+    pub fn with_id(id: u32) -> Self {
+        Self(id)
+    }
+}
+
+#[allow(unused)]
+#[derive(Copy, Clone, Debug, Eq, Hash, PartialEq)]
+pub enum DropReason {
+    InternalFailure,      /* catch-all for internal issues */
+    NotEthernet,          /* could not get eth header */
+    NotIp,                /* could not get IP header - maybe it's not ip */
+    MacNotForUs,          /* frame is not broadcast nor for us */
+    InterfaceDetached,    /* interface has not been attached to any VRF */
+    InterfaceAdmDown,     /* interface is admin down */
+    InterfaceOperDown,    /* interface is oper down : no link */
+    InterfaceUnknown,     /* the interface cannot be found */
+    InterfaceUnsupported, /* the operation is not supported on the interface */
+    VrfUnknown,           /* the vrf does not exist */
+    NatOutOfResources,    /* can't do NAT due to lack of resources */
+    RouteFailure,         /* missing routing information */
+    HopLimitExceeded,     /* TTL / Hop count was exceeded */
+    Filtered,             /* The packet was administratively filtered */
+    Unhandled,            /* there exists no support to handle this type of packet */
+    Delivered,            /* the packet buffer was delivered by the NF - e.g. for xmit */
+}
+
+#[allow(unused)]
+#[derive(Debug, Default)]
+pub struct PacketMeta {
+    pub iif: InterfaceId,             /* incoming interface - set early */
+    pub oif: InterfaceId,             /* outgoing interface - set late */
+    pub is_l2bcast: bool,             /* frame is broadcast */
+    pub is_iplocal: bool,             /* frame contains an ip packet for local delivery */
```

Comment:
this seems like something that needs to be a method rather than a bool

Reply:
A method of what type?